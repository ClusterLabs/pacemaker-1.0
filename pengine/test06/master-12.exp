 <transition_graph cluster-delay="60s" stonith-timeout="60s" failed-stop-offset="INFINITY" failed-start-offset="INFINITY" batch-limit="30" transition_id="0">
   <synapse id="0">
     <action_set>
       <rsc_op id="39" operation="promote" operation_key="sf:0_promote_0" on_node="sel3" on_node_uuid="1ee1b756-2292-49bf-aed2-8ff15e8cc503">
         <primitive id="sf:0" long-id="ms-sf:sf:0" class="ocf" provider="heartbeat" type="Stateful"/>
         <attributes CRM_meta_clone="0" CRM_meta_clone_max="2" CRM_meta_clone_node_max="1" CRM_meta_master_max="1" CRM_meta_master_node_max="1" CRM_meta_role="Master" CRM_meta_stateful="true" CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <pseudo_event id="47" operation="promote" operation_key="ms-sf_promote_0"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="1">
     <action_set>
       <pseudo_event id="47" operation="promote" operation_key="ms-sf_promote_0">
         <attributes CRM_meta_clone_max="2" CRM_meta_clone_node_max="1" CRM_meta_master_max="1" CRM_meta_master_node_max="1" CRM_meta_stateful="true" CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="2" priority="1000000">
     <action_set>
       <pseudo_event id="48" operation="promoted" operation_key="ms-sf_promoted_0">
         <attributes CRM_meta_clone_max="2" CRM_meta_clone_node_max="1" CRM_meta_master_max="1" CRM_meta_master_node_max="1" CRM_meta_stateful="true" CRM_meta_timeout="20000" crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="39" operation="promote" operation_key="sf:0_promote_0" on_node="sel3" on_node_uuid="1ee1b756-2292-49bf-aed2-8ff15e8cc503"/>
       </trigger>
     </inputs>
   </synapse>
 </transition_graph>

