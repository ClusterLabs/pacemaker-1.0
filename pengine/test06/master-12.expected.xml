<?xml version="1.0" encoding="UTF-8"?>
<cib admin_epoch="0" epoch="13" num_updates="99" dc-uuid="5d98a2ff-f9bf-45be-964b-27e4246c0085" have-quorum="true" remote-tls-port="0" validate-with="pacemaker-1.0">
   <configuration>
     <crm_config>
       <cluster_property_set id="cib-bootstrap-options"/>
     </crm_config>
     <nodes>
       <node id="1ee1b756-2292-49bf-aed2-8ff15e8cc503" uname="sel3" type="normal"/>
       <node id="5d98a2ff-f9bf-45be-964b-27e4246c0085" uname="sel4" type="normal"/>
     </nodes>
     <resources>
       <master id="ms-drbd0"><meta_attributes id="master_slave-ms-drbd0.meta"/><meta_attributes id="meta_attributes"><nvpair id="nvpair" name="clone-max" value="2"/><nvpair id="nvpair" name="clone-node-max" value="1"/><nvpair id="nvpair" name="master-max" value="1"/><nvpair id="nvpair" name="master-node-max" value="1"/><nvpair id="nvpair" name="notify" value="yes"/><nvpair id="nvpair" name="globally-unique" value="false"/></meta_attributes><primitive id="drbd0" class="ocf" provider="heartbeat" type="drbd"><meta_attributes id="primitive-drbd0.meta"/>
           <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="drbd_resource" value="drbd0"/></instance_attributes>
         </primitive></master>
       <master id="ms-sf"><meta_attributes id="master_slave-ms-sf.meta"/><meta_attributes id="meta_attributes"><nvpair id="nvpair" name="clone-max" value="2"/><nvpair id="nvpair" name="clone-node-max" value="1"/><nvpair id="nvpair" name="master-max" value="1"/><nvpair id="nvpair" name="master-node-max" value="1"/></meta_attributes><primitive id="sf" class="ocf" provider="heartbeat" type="Stateful"><meta_attributes id="primitive-sf.meta"/>
           <instance_attributes id="instance_attributes"/>
         </primitive></master>
       <primitive id="fs0" class="ocf" provider="heartbeat" type="Filesystem"><meta_attributes id="primitive-fs0.meta"/>
         <meta_attributes id="meta_attributes"/>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="fstype" value="ext3"/><nvpair id="nvpair" name="directory" value="/drbd"/><nvpair id="nvpair" name="device" value="/dev/drbd0"/></instance_attributes>
       </primitive>
     </resources>
     <constraints>
       <rsc_location id="rsc_location_group_0" rsc="ms-sf">
         <rule id="preferred_location_group_0" role="Master" score="100">
           <expression id="expression" attribute="#uname" operation="eq" value="sel3"/>
         </rule>
       </rsc_location>
       <rsc_location id="rsc_location_group_1" rsc="ms-drbd0">
         <rule id="preferred_location_group_1" role="Master" score="100">
           <expression id="expression" attribute="#uname" operation="eq" value="sel3"/>
         </rule>
       </rsc_location>
       <rsc_order id="drbd0_before_fs0" first="ms-drbd0" then="fs0" then-action="start" first-action="promote" score="0"/>
       <rsc_order id="sf_before_fs0" first="ms-sf" then="fs0" then-action="start" first-action="promote" score="0"/>
       <rsc_colocation id="fs0_on_drbd0" rsc="fs0" with-rsc="ms-drbd0" with-rsc-role="Master" score="INFINITY"/>
     </constraints>
   </configuration>
   <status>
     <node_state id="5d98a2ff-f9bf-45be-964b-27e4246c0085" uname="sel4" crmd="online" shutdown="0" in_ccm="true" ha="active" join="member" expected="member">
       <transient_attributes id="transient_attributes.auto-1">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="default_ping_set" value="100"/><nvpair id="nvpair" name="probe_complete" value="true"/></instance_attributes>
         <instance_attributes id="instance_attributes"><nvpair name="master-drbd0:1" id="nvpair" value="75"/></instance_attributes>
       </transient_attributes>
       <lrm id="lrm.auto-1">
         <lrm_resources id="lrm_resources">
           <lrm_resource id="sf:1" type="Stateful" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="sf-1_monitor_0" operation="monitor" transition-key="10:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;10:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="4" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8"/>
             <lrm_rsc_op id="sf-1_start_0" operation="start" transition-key="33:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;33:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="7" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8"/>
           </lrm_resource>
           <lrm_resource id="sf:0" type="Stateful" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="sf-0_monitor_0" operation="monitor" transition-key="9:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;9:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="3" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8"/>
           </lrm_resource>
           <lrm_resource id="fs0" type="Filesystem" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="fs0_monitor_0" operation="monitor" transition-key="11:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;11:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="5" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="fce792c58e596a12a9b098046d505dc8"/>
           </lrm_resource>
           <lrm_resource id="drbd0:0" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-0_monitor_0" operation="monitor" transition-key="8:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;8:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="2" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
           </lrm_resource>
           <lrm_resource id="drbd0:1" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-1_monitor_0" operation="monitor" transition-key="5:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;5:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="6" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-1_start_0" operation="start" transition-key="5:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;5:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="8" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-1_post_notify_start_0" operation="notify" transition-key="56:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;56:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="9" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-1_pre_notify_promote_0" operation="notify" transition-key="66:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;66:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="10" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-1_post_notify_promote_0" operation="notify" transition-key="67:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;67:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="11" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
           </lrm_resource>
         </lrm_resources>
       </lrm>
     </node_state>
     <node_state id="1ee1b756-2292-49bf-aed2-8ff15e8cc503" uname="sel3" crmd="online" in_ccm="true" ha="active" join="member" expected="member" shutdown="0">
       <transient_attributes id="transient_attributes.auto-2">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="default_ping_set" value="100"/><nvpair id="nvpair" name="probe_complete" value="true"/></instance_attributes>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="master-drbd0:0" value="75"/></instance_attributes>
       </transient_attributes>
       <lrm id="lrm.auto-2">
         <lrm_resources id="lrm_resources">
           <lrm_resource id="sf:1" type="Stateful" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="sf-1_monitor_0" operation="monitor" transition-key="5:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;5:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="4" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8"/>
           </lrm_resource>
           <lrm_resource id="sf:0" type="Stateful" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="sf-0_monitor_0" operation="monitor" transition-key="4:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;4:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="3" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8"/>
             <lrm_rsc_op id="sf-0_start_0" operation="start" transition-key="32:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;32:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="7" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="f2317cad3d54cec5d7d7aa7d0bf35cf8"/>
           </lrm_resource>
           <lrm_resource id="fs0" type="Filesystem" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="fs0_monitor_0" operation="monitor" transition-key="6:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;6:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="5" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="fce792c58e596a12a9b098046d505dc8"/>
             <lrm_rsc_op id="fs0_start_0" operation="start" transition-key="46:4:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;46:4:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="13" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="fce792c58e596a12a9b098046d505dc8"/>
           </lrm_resource>
           <lrm_resource id="drbd0:0" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-0_monitor_0" operation="monitor" transition-key="3:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;3:0:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="2" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-0_start_0" operation="start" transition-key="4:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;4:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="8" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-0_post_notify_start_0" operation="notify" transition-key="53:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;53:2:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="9" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-0_pre_notify_promote_0" operation="notify" transition-key="58:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;58:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="10" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-0_promote_0" operation="promote" transition-key="6:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;6:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="11" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
             <lrm_rsc_op id="drbd0-0_post_notify_promote_0" operation="notify" transition-key="59:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:0;59:3:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="12" crm_feature_set="1.0.9" rc-code="0" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
           </lrm_resource>
           <lrm_resource id="drbd0:1" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="drbd0-1_monitor_0" operation="monitor" transition-key="3:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" transition-magic="0:7;3:1:ab83eef1-2912-43cf-8a88-910f72f1f57b" call-id="6" crm_feature_set="1.0.9" rc-code="7" op-status="0" interval="0" op-digest="1243523f1dae58b4aafa2650a7f3d441"/>
           </lrm_resource>
         </lrm_resources>
       </lrm>
     </node_state>
   </status>
 </cib>
